src/EnergyPlus/UnitarySystem.cc

```diff
@@ -242,7 +242,10 @@ namespace UnitarySystems {
         // MassFlowRateMaxAvail issues are impeding non-VAV air loop equipment by limiting air flow
         // temporarily open up flow limits while simulating, and then set this same value at the INLET after this parent has simulated
         Real64 tempMassFlowRateMaxAvail = state.dataLoopNodes->Node(this->AirInNode).MassFlowRateMaxAvail;
-        state.dataLoopNodes->Node(this->AirInNode).MassFlowRateMaxAvail = this->m_DesignMassFlowRate;
+        // this is not working for CoilSystem simulated with UnitarySystem. Try to protect when this happens.
+        if (AirLoopNum > 0 && this->m_ControlType != ControlType::Setpoint) {
+            state.dataLoopNodes->Node(this->AirInNode).MassFlowRateMaxAvail = this->m_DesignMassFlowRate;
+        }
```

Comment:
Something is stomping on MaxAvail. Even with this change some files show evidence of issues with flow rate.

E.g., this new warning in AirflowNetwork_MultiAirLoops seems to show an issue with flow rate.

    +   ** Warning ** The mass flow rate difference is found between System Node = 'ZONE EQUIPMENT INLET NODE' and AFN Link = 'MAIN LINK 1'.
    +   **   ~~~   ** The system node max mass flow rate = 0.753 kg/s. The AFN node mass flow rate = 1.364 kg.s.
        ** Warning ** The mass flow rate difference is found between System Node = 'ZONE 1 INLET NODE ATINLET' and AFN Link = 'ZONE1SUPPLY2LINK'.
        **   ~~~   ** The system node max mass flow rate = 0.753 kg/s. The AFN node mass flow rate = 0.764 kg.s.
        ** Warning ** Please adjust the rate of Maximum Air Flow Rate field in the terminal objects or duct pressure resistance.